bin/plugin/lib/milestone.js

```diff
@@ -0,0 +1,96 @@
+/** @typedef {import('@octokit/rest')} GitHub */
+/** @typedef {import('@octokit/rest').IssuesListForRepoResponseItem} IssuesListForRepoResponseItem */
+/** @typedef {import('@octokit/rest').IssuesListMilestonesForRepoResponseItem} OktokitIssuesListMilestonesForRepoResponseItem */
+
+/**
+ * @typedef {"open"|"closed"|"all"} IssueState
+ */
+
+/**
+ * Given a SemVer-formatted version string, returns an assumed milestone title
+ * associated with that version.
+ *
+ * @see https://semver.org/
+ *
+ * @param {string} version Version string.
+ *
+ * @return {string} Milestone title.
+ */
+function getVersionMilestoneTitle( version ) {
+	const [ major, minor ] = version.split( '.' );
+
+	return `Gutenberg ${ major }.${ minor }`;
+}
+
+/**
+ * Returns a promise resolving to a milestone by a given title, if exists.
+ *
+ * @param {GitHub} octokit Initialized Octokit REST client.
+ * @param {string} owner   Repository owner.
+ * @param {string} repo    Repository name.
+ * @param {string} title   Milestone title.
+ *
+ * @return {Promise<OktokitIssuesListMilestonesForRepoResponseItem|void>} Promise resolving to milestone, if exists.
+ */
+async function getMilestoneByTitle( octokit, owner, repo, title ) {
```

Comment:
Do we need to be authenticated for these calls? Asking to see whether we could avoid the "octokit" argument  entirely.

Reply:
> Do we need to be authenticated for these calls? Asking to see whether we could avoid the "octokit" argument entirely.

Technically no, but there are pretty restrictive limits on the number of REST API calls which can be made anonymously.

Related: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/release.md#writing-the-release-post-and-changelog

>`--token <token>`: Provide a [GitHub personal access token](https://github.com/settings/tokens) for authenticating requests. This should only be necessary if you run the script frequently enough to been blocked by [rate limiting](https://developer.github.com/v3/#rate-limiting).

But I was thinking similarly, at least since the implementation here for the release tool doesn't make use of the token. It depends if we'd rather refactor toward using the same client initialized later in the process (after the user is prompted for personal access token).